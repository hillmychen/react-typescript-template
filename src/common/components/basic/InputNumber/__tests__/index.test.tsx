import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'
import { InputNumber } from '../index'

describe('InputNumber', () => {
  test('onChange Event', done => {
    const { asFragment, getByPlaceholderText } = render(<InputNumber placeholder="number" onChange={handlerOnChange} />)
    const input = getByPlaceholderText('number') as HTMLInputElement
    fireEvent.change(input, { target: { value: 3 } })
    function handlerOnChange(value: number | undefined) {
      expect(value).toBe(3)
      expect(asFragment()).toMatchSnapshot()
      done()
    }
  })

  test('test min', () => {
    const { asFragment, getByPlaceholderText } = render(<InputNumber placeholder="number" value={5} min={10} />)
    const input = getByPlaceholderText('number') as HTMLInputElement
    expect(input.value).toBe('10')
    expect(asFragment()).toMatchSnapshot()
  })

  test('test max', () => {
    const { asFragment, getByPlaceholderText } = render(<InputNumber placeholder="number" value={12} max={10} />)
    const input = getByPlaceholderText('number') as HTMLInputElement
    expect(input.value).toBe('10')
    expect(asFragment()).toMatchSnapshot()
  })

  test('test precision', () => {
    const { asFragment, getByPlaceholderText } = render(<InputNumber placeholder="number" precision={4} value={12} max={10} />)
    const input = getByPlaceholderText('number') as HTMLInputElement
    expect(input.value).toBe('10.0000')
    expect(asFragment()).toMatchSnapshot()
  })
})
