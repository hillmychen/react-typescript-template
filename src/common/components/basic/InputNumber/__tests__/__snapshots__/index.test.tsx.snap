// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber onChange Event 1`] = `
<DocumentFragment>
  <div
    class="ant-input-number normalInputNumber"
  >
    <div
      class="ant-input-number-input-wrap"
    >
      <input
        class="ant-input ant-input-number-input"
        placeholder="number"
        type="text"
        value="3"
      />
    </div>
    <div
      class="ant-input-number-handler-wrap"
    >
      <span
        aria-label="Increase Value"
        class="ant-input-number-handler ant-input-number-handler-up"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="up"
          class="anticon anticon-up ant-input-number-handler-up-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="up"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
            />
          </svg>
        </span>
      </span>
      <span
        aria-label="Decrease Value"
        class="ant-input-number-handler ant-input-number-handler-down"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-input-number-handler-down-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InputNumber test max 1`] = `
<DocumentFragment>
  <div
    class="ant-input-number normalInputNumber"
  >
    <div
      class="ant-input-number-input-wrap"
    >
      <input
        class="ant-input ant-input-number-input"
        placeholder="number"
        type="text"
        value="10"
      />
    </div>
    <div
      class="ant-input-number-handler-wrap"
    >
      <span
        aria-label="Increase Value"
        class="ant-input-number-handler ant-input-number-handler-up ant-input-number-handler-up-disabled"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="up"
          class="anticon anticon-up ant-input-number-handler-up-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="up"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
            />
          </svg>
        </span>
      </span>
      <span
        aria-label="Decrease Value"
        class="ant-input-number-handler ant-input-number-handler-down"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-input-number-handler-down-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InputNumber test min 1`] = `
<DocumentFragment>
  <div
    class="ant-input-number normalInputNumber"
  >
    <div
      class="ant-input-number-input-wrap"
    >
      <input
        class="ant-input ant-input-number-input"
        placeholder="number"
        type="text"
        value="10"
      />
    </div>
    <div
      class="ant-input-number-handler-wrap"
    >
      <span
        aria-label="Increase Value"
        class="ant-input-number-handler ant-input-number-handler-up"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="up"
          class="anticon anticon-up ant-input-number-handler-up-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="up"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
            />
          </svg>
        </span>
      </span>
      <span
        aria-label="Decrease Value"
        class="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-input-number-handler-down-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InputNumber test precision 1`] = `
<DocumentFragment>
  <div
    class="ant-input-number normalInputNumber"
  >
    <div
      class="ant-input-number-input-wrap"
    >
      <input
        class="ant-input ant-input-number-input"
        placeholder="number"
        type="text"
        value="10.0000"
      />
    </div>
    <div
      class="ant-input-number-handler-wrap"
    >
      <span
        aria-label="Increase Value"
        class="ant-input-number-handler ant-input-number-handler-up ant-input-number-handler-up-disabled"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="up"
          class="anticon anticon-up ant-input-number-handler-up-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="up"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
            />
          </svg>
        </span>
      </span>
      <span
        aria-label="Decrease Value"
        class="ant-input-number-handler ant-input-number-handler-down"
        role="button"
        tabindex="-1"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-input-number-handler-down-inner"
          role="img"
          style="font-size: 8px;"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
